build/cmake/MongoC-Warnings.cmake

```diff
@@ -0,0 +1,46 @@
+#[[
+   Define additional compile options, conditional on the compiler being used.
+   Each option should be prefixed by `gnu-like:` or `msvc:`. Those options will be
+   enabled for GCC/Clang or MSVC respectively.
+
+   These options are attached to the source directory and its children.
+]]
+function (mongoc_add_platform_compile_options)
+   foreach (opt IN LISTS ARGV)
+      if (NOT opt MATCHES "^(gnu-like|msvc):(.*)")
+         message (SEND_ERROR "Invalid option '${opt}' (Should be prefixed by 'msvc:' or 'gnu-like:'")
+         continue ()
+      endif ()
+      set (is_gnu_like "$<C_COMPILER_ID:GNU,Clang,AppleClang>")
+      set (is_msvc "$<C_COMPILER_ID:MSVC>")
+      if (CMAKE_MATCH_1 STREQUAL "gnu-like")
+         add_compile_options ("$<${is_gnu_like}:${CMAKE_MATCH_2}>")
```

Comment:
```
CMake Error at build/cmake/MongoC-Warnings.cmake:17 (add_compile_options):
   Error evaluating generator expression:
     $<C_COMPILER_ID:GNU,Clang,AppleClang>
   $<C_COMPILER_ID> expression requires one or zero parameters.
```

Evergreen tasks appear to be failing during CMake configuration due to invalid generator expression evaluation given definition of `is_gnu_like` on line 14. `$<C_COMPILER_ID:compiler_ids>` (plural "ids") appears to be a [CMake 3.15 and newer feature](https://cmake.org/cmake/help/v3.15/manual/cmake-generator-expressions.7.html#variable-queries). Either the `CMAKE_MINIMUM_REQUIRED` version should be bumped to `3.15`, or the condition should be rewritten, e.g. something like:

```cmake
set (is_gnu_like "$<IN_LIST:$<C_COMPILER_ID>,GNU;Clang;AppleClang>")
```

Reply:
Addressed by https://github.com/mongodb/mongo-c-driver/pull/890/commits/5c1f02724b4dde8fd46f639272b6181cf226b019.